pkg/api/routes.go

```diff
@@ -127,11 +127,17 @@ func (rh *RouteHandler) SetupRoutes() {
 		prefixedDistSpecRouter.Use(DistSpecAuthzHandler(rh.c))
 	}
 
+	clusterRouteProxy := ClusterProxy(rh.c.Config)
+
 	// https://github.com/opencontainers/distribution-spec/blob/main/spec.md#endpoints
 	{
 		prefixedDistSpecRouter.HandleFunc(fmt.Sprintf("/{name:%s}/tags/list", zreg.NameRegexp.String()),
-			getUIHeadersHandler(rh.c.Config, http.MethodGet, http.MethodOptions)(
-				applyCORSHeaders(rh.ListTags))).Methods(http.MethodGet, http.MethodOptions)
+			clusterRouteProxy(
```

Comment:
The reason for wrapping the handler in this manner is because we would like to use `mux.Vars` inside the Proxy code to get the repo name.

Attaching the middleware directly to the router does not allow us to use the variables in this manner.
Ref: https://stackoverflow.com/a/43980966

Reply:
This does mean that every endpoint that would be proxied would need to be wrapped in this manner. Not too sure if that's a good design.

We could apply the proxy as a middleware at the router level, but we'd have to do apply regex manually to figure out the repo name correctly.

Reply:
But let's tackle that later. First let's get an end-to-end test going.

Reply:
The main objection is that we don't want to "pollute" pkg/api/routes.go with clustering if possible - performance issues may force our hand though.